### PR TITLE
Input box > Theme service optionally provided

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**
Related: https://github.com/blackbaud/skyux-docs-tools/pull/137

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-input-box-optional-theme-svc](https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-input-box-optional-theme-svc)

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-input-box-optional-theme-svc](https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-input-box-optional-theme-svc)

File: src/app/public/modules/input-box/input-box.component.spec.ts [hunk 1]
```diff
@@ -144,26 +162,24 @@ describe('Input box component', () => {
     });
   });
 
-  it('should render the label and input elements in the expected locations', () => {
+  it('should handle SkyThemeService not provided', () => {
+    TestBed.overrideProvider(SkyThemeService, {
```
Comment (Blackbaud-PaulCrowder) [inline src/app/public/modules/input-box/input-box.component.spec.ts:166]: This is going to ensure that the component handles an undefined `SkyThemeService`, but it doesn't test the case where Angular has no provider for `SkyThemeService`. I think I'd maybe flip these two tests where the `TestBed` doesn't have `SkyThemeService` provided at all, but the test where we ensure providing it has the desired effect uses `overrideComponent` to inject `SkyThemeService`.

Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/input-box/input-box.component.spec.ts]: I had to wrap the default and modern tests in their own `describe` blocks and create separate `TestBed`s to get it to work. I also put the duplicate testing module configuration in an `InputBoxFixturesModule`.

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-input-box-optional-theme-svc](https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-input-box-optional-theme-svc)

**Review by Blackbaud-PaulCrowder — APPROVED** [review]

**Pull request merged**